cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,256 @@
+                self.changes.append(
+                    Change(
+                        target_section.key,
+                        target_section.label,
+                        param.key,
+                        base_value,
+                        param.value,
+                        param.get_update_policy(),
+                    )
+                )
+
+    def _compare_params(self, section_key, param_key, base_value, target_value):
+        """
+        Compare the specified base configuration parameter with the corresponding one in target configuration.
+
+        The way the comparison is performed depends on the availability of a custom comparator for the specified
+        parameter. If no custom comparator is available, a default equality comparison is performed.
+
+        :param section_key: The section key
+        :param param_key: The parameter key
+        :param base_value: The value of the parameter in the base configuration
+        :param target_value: The value of the parameter in the target configuration
+        :return: True if the values can be considered equal, False otherwise
+        """
+        comparator = ConfigPatch.PARAM_COMPARATORS.get((section_key.lower(), param_key.lower()))
+        if comparator:
+            result = comparator(base_value, target_value)
+        else:
+            result = base_value == target_value
+        return result
+
+    def _remove_ignored_sections(self, config):
+        for section_key in ConfigPatch.IGNORED_SECTIONS:
+            config.remove_section(section_key)
+
+    @property
+    def update_policy_level(self):
+        """
+        Get the max update policy level of the ConfigPatch.
+
+        This method provides a static indication of the actual applicability of the patch and it's meant for testing
+        purposes. The real applicability of a patch, which involves dynamic condition checks, must be performed
+        by calling the check() method.
+        """
+        return (
+            max(change.update_policy.level for change in self.changes)
+            if len(self.changes)
+            else UpdatePolicy.SUPPORTED.level
+        )
+
+    def _create_default_section(self, config, section):
+        """
+        Create a section of same type of the provided section, with all default parameter values.
+
+        The purpose of this operation is to allow sections comparison when a section is missing in base or target
+        configuration.
+
+        :param config: The configuration lacking the section
+        :param section: The section to create a default copy of
+        """
+        section_definition = section.definition
+        section_type = section_definition.get("type")
+        default_section = section_type(
+            section_definition=section_definition, pcluster_config=config, section_label=section.label
+        )
+
+        return default_section
+
+    def _get_config_section(self, config, section):
```

Comment:
Same as before. It's a static method, the `self` object is not used here, I'd move it outside the `ConfigPatch` object.

I'd move it in the `pcluster_config.py` module.

Reply:
As before, I think it logically fits inside this class; I can't figure out a valid use case for having this behaviour available globally or in the PClusterConfig.

For the same reason I'd say I'm even not convinced about making it static since it's really intended to be used only from inside a ConfigPatch. I would make it private with the "__" prefix.